### ISSUE TITLE
Domain whitelist suggestion only works correctly on simple TLDs

### ISSUE THREAD

**Issue by MisaelK** [mrdokenny/Cookie-AutoDelete#243]
### Describe the Bug/Feature

When browsing any site with a multipart TLD (that is, one that is formed by more than one name, like `co.uk` or `com.ar`), the domain whitelist/greylist suggestion box fails to provide the same type of options available in simple TLD sites. For instance:

`images.google.com` shows:
* `images.google.com`
* `*.google.com`
* `*.images.google.com`

`images.google.com.ar` shows:
* `*.com.ar`
* `images.google.com.ar`
* `*.images.google.com.ar`

I'm using a `.com.ar` site as an example, but this apparently happens with any multipart TLD (like `co.uk`, `co.jp`, `com.br`, `com.mx`, etc), while working correctly on simple TLD (like `.com`, `.de`, etc).

### Screenshot in which the bug can be seen

[Bug screenshot](https://i.imgur.com/d8vwHLw.png)

[Expected behavior screenshot](https://i.imgur.com/C4zmpQT.png)

### Steps for anyone to reproduce the bug
1. Visit any site with a multipart TLD
2. Click the Cookie AutoDelete menu toolbar button
3. Expect to see the same suggestions available in sites ending with a simple TLD

### Your settings 
- OS/version: Windows 10 Enterprise 2015 LTSB x64
- Browser/version: Firefox 56.0.2 (64-bits)
- Cookie AutoDelete version: 2.0.1

**Comment by practik** [comment]
@MisaelK, just to clarify, the problem is that first suggestion for `images.google.com.ar` is `*.com.ar`, but it should be `*.google.com.ar`, correct?

https://github.com/gorhill/publicsuffixlist.js might be a useful tool for @mrdokenny here.

**Comment by MisaelK** [comment]
@practik Yes, that is exactly the problem.

**Comment by drcheap** [comment]
Technically speaking, **com.ar** is a 2nd level domain under the **ar ccTLD**.  But that's just being pedantic :)

Really it seems that the core issue is not all levels are being enumerated, 
`images.google.com.ar`
`*.images.google.com.ar`
`*.google.com.ar                <--- the one that is missing`
`*.com.ar`

For fun I created a [5 level deep site](http://five.level.test.drcheap.com), and still only get 3 menu options for grey/white list, when perhaps it should be 5 or 6?
`five.level.test.drcheap.com`
`*.five.level.test.drcheap.com`
`*.level.test.drcheap.com     <-- missing`
`*.test.drcheap.com     <-- missing`
`*.drcheap.com`
`*.com     <-- okay, maybe the TLD shouldn't be a choice`

Maybe doing **all** the levels is a bit much, so just doing the first 3 (which is really current, wildcard below current, and wildcard at current level) is enough.  I guess what I'm getting at is there is a way to do this that doesn't require maintaining and/or depending on an ever-changing list of TLDs.  

Also, IMO the order should always be most-specific to least-specific as I showed them here, but that's a different "issue" I suppose.
